### PR TITLE
Projects

### PR DESCRIPTION
**Pull request by mkasztelnik**
In the scope of this PR following changes were made:
  * `Order` was renamed to `ProjectItem`
  * `OrderChange` was renamed to `ProjectItemChange`
  * New model `Project` was introduced. The user is required to select a project while requesting service. Underneath model relations were changed (from `user has many project_items` into `user has many projects`, `project has many project_items`)

Request service form was updated and project select input was added:
select-project

In the "My Services" view services are grouped using project:
projects

### CODE REVIEW

**Comment by roksanaer** [comment]
content of the PR looks good to me.
but, not sure which branch u've used to create this one, while testing:
ordering button on the service detailed view is not visible until i logged in
after loggin in, i click the order button and it redirects me straight to:
https://marketplace-2.docker-fid.grid.cyf-kr.edu.pl/orders/3
(detailed view of the order)
used chrome to test

**Comment by roksanaer** [comment]
same applies to the firefox. so - the functionality to choose a project is not accessible

**Comment by roksanaer** [comment]
> after loggin in, i click the order button and it redirects me straight to:
> https://marketplace-2.docker-fid.grid.cyf-kr.edu.pl/orders/3
check where the order button is leading to

**Comment by mkasztelnik** [comment]
@roksanaer there is something wrong with docker deploy. It looks like not PR code is there. Please download code locally and try again.

@wziajka can you take a look at docker deploy?

**Comment by mkasztelnik** [comment]
travis failures (haml-lint) are not connected with introduced changes.

**Comment by wziajka** [comment]
Deploy fixed. Try https://marketplace-12.docker-fid.grid.cyf-kr.edu.pl/

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 class CartsController < ApplicationController
```
Comment (kmarszalek) [inline app/controllers/carts_controller.rb]: why you didnt change cotroller name to eliminate 'Cart' from code?

Comment (kmarszalek) [inline app/controllers/carts_controller.rb]: why didnt you change name to eliminate 'Cart' from code

Comment (mkasztelnik) [reply app/controllers/carts_controller.rb]: The scope of this PR is to introduce Project, not to remove Cart

File: app/views/affiliation_mailer/verification.text.haml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Dear #{@user.full_name},
 \
-You just created new affiliation. To use it during ordering process it need
+You just created new affiliation. To use it during project_iteming process it need
```
Comment (kmarszalek) [inline app/views/affiliation_mailer/verification.text.haml]: maybe "project_item/items ordering/requesting"

File: app/views/affiliation_mailer/verification.html.haml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 %p Dear #{@user.full_name},
 %p
-  You just created new affiliation. To use it during ordering process it need
+  You just created new affiliation. To use it during project_iteming process it need
```
Comment (kmarszalek) [inline app/views/affiliation_mailer/verification.html.haml]: maybe "project_item/items ordering/requesting"

File: app/views/home/_home-view.html.haml [hunk 4]
```diff
@@ -42,31 +42,31 @@
...
         .col-md-2.mt-4
-          %a.area.border.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
+          %a.area.bproject_item.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
             %h3
               Chemistry
             %span.more-link.position-absolute{ href: "" }
               %i.fas.fa-angle-right
         .col-md-2.mt-4
-          %a.area.border.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
+          %a.area.bproject_item.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
```
Comment (kmarszalek) [inline app/views/home/_home-view.html.haml]: bproject

File: app/views/playground/home-view.html.haml [hunk 5]
```diff
@@ -42,31 +42,31 @@
...
         .col-md-2.mt-4
-          %a.area.border.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
+          %a.area.bproject_item.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
             %h3
               Chemistry
             %span.more-link.position-absolute{ href: "" }
               %i.fas.fa-angle-right
         .col-md-2.mt-4
-          %a.area.border.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
+          %a.area.bproject_item.bg-white.d-flex.align-items-center.justify-content-center.position-relative.rounded{ href: "" }
```
Comment (kmarszalek) [inline app/views/playground/home-view.html.haml]: bproject x5

File: app/views/profiles/affiliations/_affiliation.html.haml [hunk 6]
```diff
@@ -1,4 +1,4 @@
-.card.border.rounded.mb-2.pb-2.pt-2
+.card.bproject_item.rounded.mb-2.pb-2.pt-2
```
Comment (kmarszalek) [inline app/views/profiles/affiliations/_affiliation.html.haml]: bproject_item

File: db/migrate/20181017195440_create_projects.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
```
Comment (kmarszalek) [inline db/migrate/20181017195440_create_projects.rb]: ?

File: app/views/playground/static-tabs.html.haml [hunk 8]
```diff
@@ -52,7 +52,7 @@
             %a{ href: "" }SLA
           %li
             %a{ href: "" }Access policy
-  #order.tab-pane.container{ "aria-labelledby" => "profile-tab", role: "tabpanel" }
+  #project_item.tab-pane.container{ "aria-labelledby" => "profile-tab", role: "tabpanel" }
```
Comment (kmarszalek) [inline app/views/playground/static-tabs.html.haml]: project_item is not lisp-case

File: app/views/services/_order.html.haml [hunk 9]
```diff
@@ -1,4 +1,4 @@
-#order.tab-pane.pt-2.container{ "aria-labelledby" => "profile-tab", role: "tabpanel" }
+#project_item.tab-pane.pt-2.container{ "aria-labelledby" => "profile-tab", role: "tabpanel" }
```
Comment (kmarszalek) [inline app/views/services/_order.html.haml]: [W] IdNames: `id` attribute must be in lisp-case

File: app/controllers/carts_controller.rb (hunk 1 above)
Comment (kmarszalek) [reply app/controllers/carts_controller.rb]: ok

File: db/migrate/20181017195440_create_projects.rb (hunk 7 above)
Comment (mkasztelnik) [reply db/migrate/20181017195440_create_projects.rb]: Added `frozen_string_literal` directive into all migration files.

Comment (mkasztelnik) [reply db/migrate/20181017195440_create_projects.rb]: Added `frozen_string_literal` directive into all migration files.

Comment (mkasztelnik) [reply db/migrate/20181017195440_create_projects.rb]: `frozen_string_literal` directive was added into all migration files.

Comment (mkasztelnik) [reply db/migrate/20181017195440_create_projects.rb]: `frozen_string_literal` directive was added to all migration files.

Comment (mkasztelnik) [reply db/migrate/20181017195440_create_projects.rb]: `frozen_string_literal` was added to all migration files.

**Pull request merged**